src/math.cpp

```diff
@@ -8,7 +8,16 @@ namespace openre::math
     // 0x00451710
     static int rsin(int32_t angle)
     {
-        return interop::call<int32_t, int32_t>(0x00451710, angle);
+        const uint32_t idx = angle & 0x7FF;
+        const uint32_t sign = (angle & 0x800) != 0 ? -1 : 1;
+
+        // cosine
+        if (idx >= 1024)
+        {
+            return sign * *((uint16_t*)0x0000527270 - idx);
```

Comment:
Should this be added to gGameTable, or since it is constant, replicated in the C code?

Reply:
We also have a third option: figure out where the cosine table starts and add it to the gGameTable. With that table, we can implement logic to calculate and resolve the index. However, it might be simpler to move the address to a constant.

What do you prefer?

Reply:
I think there is one table of numbers that is used for sine, cosine, and square root.
0x526270 to 0x526C00.

The 0x527270 address will always be subtracted by at least 2048 bytes, which takes you to 0x526A70 which is 2048 bytes after the sine table.

You can therefore write some code to extract all the uint16_t between 0x526270 to 0x526C00 and build a static const C array which you can then use for these functions instead of the original memory.